src/StrawberryShake/CodeGeneration/src/CodeGeneration/Mappers/OperationDescriptorMapper.cs

```diff
@@ -23,7 +23,8 @@ public static void Map(ClientModel model, IMapperContext context)
                         var typeName = arg.Type.TypeName();
 
                         var namedTypeDescriptor =
-                            context.Types.Single(type => type.Name.EqualsOrdinal(typeName));
+                            context.Types.Single(
+                                type => type.Name.EqualsInvariantIgnoreCase(typeName));
```

Comment:
why is this the solution? There could be a type `FooBar` and a type `fooBar`.

Reply:
I didn't realize that two types could have the same (case-insensitive) name. For enums, `gender` and `Gender` results in names of `Gender` and `Gender_1`, but I see that objects/interfaces can have the same name.

I apologize. I'll have to rethink this.

I see that a lowercase object type name like `user` remains `user` in the `ObjectTypeDescriptor`, but a lowercase enum type name `gender` becomes `Gender` in the `EnumTypeDescriptor` – is this expected?

Reply:
The problem seems to be caused by the `GetClassName` usage in `DocumentAnalyzer.CollectEnumTypes.cs`, which uppercases the first letter of the enum type name.

This is then compared with the type name of the operation argument, which has not been transformed in this way, so they don't match.

Removing the call to `GetClassName` "fixes" the issue, but:

1. The generated code is changed, for example from `enum People_constraint` to `enum people_constraint` (but neither of these is ideal anyway).
2. Other types are also using `GetClassName`, f.e. in `DocumentAnalyzer.CollectInputObjectTypes.cs` – removing these leads to additional generated code changes like `FormatCars(input.Cars)` -> `Formatcars(input.cars)`, etc.

It seems like this formatting of type names should happen right at the end, just before or while the generated code is being written. This is potentially a much larger change though.